third_party/miniz.c

```diff
@@ -2850,17 +2850,44 @@ void *tdefl_write_image_to_png_file_in_memory(const void *pImage, int w, int h,
   #include <sys/stat.h>
 
   #if defined(_MSC_VER) || defined(__MINGW64__)
+    #include <Windows.h>
+    #include <stringapiset.h>
+    #include <wchar.h>
+    static wchar_t* utf8_to_wchar(const char* inStr)
+    {
+      if (inStr == NULL)
+        return NULL;
+      int wchars_count = MultiByteToWideChar(CP_UTF8, 0, inStr, -1, NULL, 0);
+      wchar_t* wstr = malloc(sizeof(wchar_t)*wchars_count);
+      if (MultiByteToWideChar(CP_UTF8, 0, inStr, -1, wstr, wchars_count) == 0)
+      {
+        wstr[0] = L'\0';
+      }
+      return wstr;
+    }
+
     static FILE *mz_fopen(const char *pFilename, const char *pMode)
     {
       FILE* pFile = NULL;
-      fopen_s(&pFile, pFilename, pMode);
+      wchar_t* pFilenameW = utf8_to_wchar(pFilename);
+      wchar_t* pModeW = utf8_to_wchar(pMode);
+     (void)_wfopen_s(&pFile, pFilenameW, pModeW);
+      free(pFilenameW);
+      free(pModeW);
       return pFile;
     }
+    
     static FILE *mz_freopen(const char *pPath, const char *pMode, FILE *pStream)
     {
       FILE* pFile = NULL;
-      if (freopen_s(&pFile, pPath, pMode, pStream))
-        return NULL;
+      wchar_t* pFilenameW = utf8_to_wchar(pPath);
+      wchar_t* pModeW = utf8_to_wchar(pMode);
+      if (_wfreopen_s(&pFile, pFilenameW, pModeW, pStream))
+      {
+        pFile = NULL;
+      }
+      free(pFilenameW);
```

Comment:
if null is passed to free, it's a no-op. 
http://www.cplusplus.com/reference/cstdlib/free/